packages/tools/lib/create-new-component/index.js

```diff
@@ -108,47 +35,93 @@ const getLibraryName = packageName => {
 	return packageName.substr("webcomponents-".length);
 };
 
+// String manipulation
 const camelToKebabCase = string => string.replace(/([a-z])([A-Z])/g, "$1-$2").toLowerCase();
-
-const packageName = getPackageName();
-const library = getLibraryName(packageName);
-
-const consoleArguments = process.argv.slice(2);
-const componentName = consoleArguments[0];
-
-if (!componentName){
-	console.error("Please enter component name.");
-	return;
+const capitalizeFirstLetter = string => string.charAt(0).toUpperCase() + string.slice(1);
+
+// Validation of user input
+const isNameValid = name => typeof name === "string" && name.match(/^[a-zA-Z][a-zA-Z0-9_-]*$/);
+
+const generateFiles = (componentName, tagName, library, packageName, isTypeScript) => {
+	const correctComponentName = capitalizeFirstLetter(componentName);
+	const filePaths = {
+		"extension": isTypeScript 
```

Comment:
the word "extension" is not semantically correct here, this is the file name, not its extension. Use for example "main"